lib/dry/validation/values.rb

```diff
@@ -45,15 +45,13 @@ def [](*args)
 
         case (key = args[0])
         when Symbol, String, Array, Hash
-          path = Schema::Path[key]
-          keys = path.to_a
+          keys = Schema::Path[key].to_a
 
           return data.dig(*keys) unless keys.last.is_a?(Array)
 
           last = keys.pop
           vals = self.class.new(data.dig(*keys))
-
-          last.map { |name| vals[name] }
+          vals.fetch_values(*last)
```

Comment:
Current test doesn't cover what's the expected value if the key doesn't exist, and following the flow looks like it should not be a problem.
If need to have a non-existent key (as nil) this would probably needed.
```suggestion
          vals.fetch_values(*last) { nil }
```

Reply:
I think we want to stick to Ruby's (poor, but "idiomatic" convention), that it should return `nil` when a key doesn't exist.